server/mcp-service.ts

```diff
@@ -71,9 +70,11 @@ export async function handleMcpPost(req: Request, res: Response): Promise<void>
     transport = session.transport;
     mcpServer = session.mcpServer;
     console.log(`  ♻️ Reusing existing transport for session: ${sessionId}`);
-  } 
-  else if (!sessionId && isInitializeRequest(req.body as JSONRPCRequest)) {
-    // New initialization request
+  }
+  // Preferring to not use isInitializeRequest here due to returning false negatives on malformed requests
+  // e.g. missing clientInfo or having an empty object for capabilities.roots
+  else if (!sessionId && req.body?.method === 'initialize') {
```

Comment:
waht does `isInitializeRequest` use?  What case did you encounter that made this switch?